core/test/mpi/base/bindings.cpp

```diff
@@ -170,23 +211,26 @@ TYPED_TEST(MpiBindings, CanPutValuesWithExclusiveLock)
     auto my_rank = comm.rank();
     auto num_ranks = comm.size();
     std::vector<TypeParam> data;
+
     if (my_rank == 0) {
         data = std::vector<TypeParam>{1, 2, 3, 4};
     } else {
         data = std::vector<TypeParam>{0, 0, 0, 0};
     }
-    auto win = window(data.data(), 4, comm);
-    if (my_rank == 0) {
-        for (auto rank = 0; rank < num_ranks; ++rank) {
-            if (rank != my_rank) {
-                win.lock(rank, 0, window::lock_type::exclusive);
-                win.put(data.data(), 4, rank, 0, 4);
-                win.flush(0);
-                win.unlock(rank);
+
+    {
+        auto win = window(data.data(), 4, comm);
+        if (my_rank == 0) {
+            for (auto rank = 0; rank < num_ranks; ++rank) {
+                if (rank != my_rank) {
+                    win.lock(rank, 0, window::lock_type::exclusive);
```

Comment:
If the most used value of `assert` is `0`, perhaps that could be the last argument to `lock` so that we normally don't have to bother with it while creating an exclusive lock.